src/ExtensionsSample/Samples/SendGridSamples.cs

```diff
@@ -20,32 +19,63 @@ public static class SendGridSamples
                 To = "{CustomerEmail}",
                 Subject = "Thanks for your order (#{OrderId})!",
                 Text = "{CustomerName}, we've received your order ({OrderId}) and have begun processing it!")]
-            SendGridMessage message)
+            out SendGridMessage message)
         {
             // You can set additional message properties here
+            message = new SendGridMessage();
+            message.Headers.Add("MyHeader", "MyValue");
         }
 
         /// <summary>
-        /// Demonstrates imperatively defining all email properties in the function body.
-        /// Also demonstrates use of a 'ref' parameter, allowing you to decide in your
-        /// method whether to send the message.
+        /// Demonstrates imperatively setting email properties inline in the function.
         /// </summary>
+        [Disable]
         public static void ProcessOrder_Imperative(
             [QueueTrigger(@"samples-orders")] Order order,
-            [SendGrid] ref SendGridMessage message)
```

Comment:
Same here - we no longer support `ref`. To conditionally send messages, you use `IAsyncCollector` which is a new addition with this replat. See the example below.